cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/realtime/sinks/RealtimeTableSink.java

```diff
@@ -16,54 +16,79 @@
-  private static final String NAME = "name";
+
+  private static final String NAME_DESC = "Name of the table. If the table does not already exist, one will be " +
+    "created.";
+  private static final String PROPERTY_SCHEMA_DESC = "Optional schema of the table as a JSON Object. If the table " +
+    "does not already exist, one will be created with this schema, which will allow the table to be explored " +
+    "through Hive.\"";
+  private static final String PROPERTY_SCHEMA_ROW_FIELD_DESC = "The name of the record field that should be used as " +
+    "the row key when writing to the table.";
+
   private RecordPutTransformer recordPutTransformer;
-  private String datasetName;
 
-  @Override
-  public void configure(StageConfigurer configurer) {
-    configurer.setDescription("Real Time Sink for CDAP Table dataset");
-    configurer.addProperty(new Property(NAME, "Name of the table. If the table does not already exist," +
-      " one will be created.", true));
-    configurer.addProperty(
-      new Property(Table.PROPERTY_SCHEMA,
-                   "Optional schema of the table as a JSON Object. If the table does not already exist," +
-                     " one will be created with this schema, which will allow the table to be explored through Hive.",
-                   false));
-    configurer.addProperty(
-      new Property(Table.PROPERTY_SCHEMA_ROW_FIELD,
-                   "The name of the record field that should be used as the row key when writing to the table.",
-                   true));
+  /**
+   * Config class for RealtimeTableSink
+   */
+  public static class TableConfig extends PluginConfig {
+    @Description(NAME_DESC)
+    private String name;
+
+    @Name(Properties.Table.PROPERTY_SCHEMA)
+    @Description(PROPERTY_SCHEMA_DESC)
+    @Nullable
+    String schemaStr;
+
+    @Name(Properties.Table.PROPERTY_SCHEMA_ROW_FIELD)
+    @Description(PROPERTY_SCHEMA_ROW_FIELD_DESC)
+    String rowField;
+
+    public TableConfig(String name, String schemaStr, String rowField) {
+      this.name = name;
+      this.schemaStr = schemaStr;
+      this.rowField = rowField;
+    }
+  }
+
+  private final TableConfig tableConfig;
+
+  public RealtimeTableSink(TableConfig tableConfig) {
+    this.tableConfig = tableConfig;
   }
 
   @Override
   public void configurePipeline(ETLStage stageConfig, PipelineConfigurer pipelineConfigurer) {
-    String tableName = stageConfig.getProperties().get(NAME);
-    String rowKeyField = stageConfig.getProperties().get(Table.PROPERTY_SCHEMA_ROW_FIELD);
-    Preconditions.checkArgument(!Strings.isNullOrEmpty(tableName), "Dataset name must be given.");
-    Preconditions.checkArgument(!Strings.isNullOrEmpty(rowKeyField), "Field to be used as rowkey must be given.");
-    pipelineConfigurer.createDataset(tableName, Table.class.getName(), DatasetProperties.builder()
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(tableConfig.name), "Dataset name must be given.");
```

Comment:
Just a thoought, we should probably just check for empty since it is guaranteed to be not null.